transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -184,7 +184,7 @@ private void notifyHealthCheck(Future<Boolean> future, Channel ch, Promise<Chann
         assert ch.eventLoop().inEventLoop();
 
         if (future.isSuccess()) {
-            if (future.getNow() == Boolean.TRUE) {
+            if (future.getNow()) {
```

Comment:
Sorry that I didn't make this change clear. Actually, my motivation is it's usually not safe to use `==` to compare Boolean since the left side may be a different object created by `new Boolean(true)`.